### PR TITLE
Support atomic ordering for LLVM load/store instructions

### PR DESCRIPTION
**Pull request by asavonic**
LLVM IR "store atomic" and "load atomic" instructions are now lowered
to the corresponding SPIR-V atomics.

Signed-off-by: Andrew Savonichev <andrew.savonichev@intel.com>

### CODE REVIEW

**Review by AlexeySachkov — APPROVED** [review]
LGTM

File: lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -1116,6 +1116,47 @@ LLVMToSPIRV::getLoopControl(const BranchInst *Branch,
...
+  case AtomicOrdering::Release:
+    return MemorySemanticsReleaseMask;
+  case AtomicOrdering::AcquireRelease:
+    return MemorySemanticsAcquireReleaseMask;
+  case AtomicOrdering::SequentiallyConsistent:
+    return MemorySemanticsSequentiallyConsistentMask;
+  default:
+    llvm_unreachable("Unsupported atomic ordering.");
+  }
+  return -1;
```
Comment (AlexeySotkin) [inline lib/SPIRV/SPIRVWriter.cpp:1134]: Can we reuse these [2 lines](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/a825cdba21306daf40725107d354a6858aed638d/lib/SPIRV/SPIRVWriter.cpp#L1549-L1550) instead?

Comment (asavonic) [reply lib/SPIRV/SPIRVWriter.cpp]: Done.

**Review by AlexeySotkin — APPROVED** [review]

**Comment by AlexeySotkin** [comment]
@asavonic could you please sign the CLA?

**Comment by asavonic** [comment]
This account is no longer associated with the Intel email address used for the commit, so I cannot sign it now.
If you cannot proceed without it, I suggest to create a new pull request.

**Comment by AlexeySotkin** [comment]
> This account is no longer associated with the Intel email address used for the commit, so I cannot sign it now.
> If you cannot proceed without it, I suggest to create a new pull request.

I'll proceed with a new pull request.

**Pull request closed without merging**